### ISSUE TITLE
bug: Gloomclaw automatically unloads

### ISSUE THREAD

**Issue by NielsH** [Immemor/RaidCore#73]
Currently, the Gloomclaw module loads correctly, but then it unloads itself straight away.

When manually forcing the module to enable, it unloads itself  again a few seconds after combat starts.

**Comment by NielsH** [comment]
Also the case for Maelstrom.

Seems it is disabled at https://github.com/NielsH/RaidCore/blob/master/RaidCore.lua#L788

**Comment by Immemor** [comment]
We should open another ticket for Maelstrom.

**Comment by Immemor** [comment]
Did you take Log ?

**Comment by NielsH** [comment]
I don't have a log for those 2 bosses. I'll see if I can clear up some disk space so I can fit the PTR client :) Access to all bosses there in DS is easier I believe, should make debugging a bit easier since the raid doesn't wait for me to try stuff out on-the-fly :p

**Comment by Immemor** [comment]
I did it also sometime, with all languages installed. You must be grouped with someone for that.

That's all.

**Comment by Immemor** [comment]
The next time, take log. That could help to understand.

**Comment by adamjedlicka** [comment]
Same for me yesterday. I was using pre #56 version of RaidCore and after firs phase RaidCore stopped working. I think it started working again few times in third phase.

**Comment by Immemor** [comment]
Well, I haven't really the choice so. A lot of thing is based on UnitCreated which is an architecture problem as UnitCreated is not mandatory.

I will start the https://github.com/NielsH/RaidCore/issues/70 ticket. I will need a least 1 week for.

**Comment by adamjedlicka** [comment]
Yeah second phase doesn't works and after that timers are off by few seconds. Everyone in the raid had this problem today.

**Comment by Immemor** [comment]
First, thanks u for your feedback.

I changed almost all the low layer to improve cpu & add log system.
And now the encounter could be disable only on Wipe event detection. That means there are probably some errors in Gloomclaw description itself.

I hope I will be able to play on this boss, this week. Otherwise, Adam, as your aren't a beginner, you could maybe forward us, your log when u meet a problem.
For that, you have the command /rclog. This interface provide you a way to send log to clipboard.
And I will be able to analyze them in a google sheet.
I wrote a wiki page about log: https://github.com/NielsH/RaidCore/wiki/How-to-use-Logs

**Comment by adamjedlicka** [comment]
Ok I'll try not to forget :)

EDIT: Dunno if this could help with timers, but there's vid of GC from yesterday: http://www.twitch.tv/wintybadass/b/656400826?t=2h17m11s

**Comment by Immemor** [comment]
interesting, indeed.

**Comment by NielsH** [comment]
Can confirm the issue, I did some debugging but not sure what happens exactly before we killed it

The timers are variable based on the `spawnTimer` variable. OnChatDC itself was properly triggered and I was able to log the messages from it within the function itself.

I think something is going wrong in the `if isPushBack or isMoveForward`  block with `OnChatDC` or within the `elseif sName == self.L["Strain Parasite"]....` bit inside `OnUnitStateChanged` because that is where waveCount is updated and those timers should be triggered.
